File: api/app/schemas/inspection.py
import uuid
from datetime import datetime

from pydantic import BaseModel, ConfigDict


class InspectionCreate(BaseModel):
    annotation_domain: str = "surface_treatment"


class InspectionUpdate(BaseModel):
    annotation_domain: str | None = None
    report_flagged: bool | None = None



class UploadUrlResponse(BaseModel):
    upload_url: str
    key: str
    expires_in: int


class ConfirmUploadRequest(BaseModel):
    key: str
    etag: str


class InspectionResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    annotation_domain: str
    inspector_id: uuid.UUID
    image_key: str | None = None
    thumbnail_key: str | None = None
    report_flagged: bool
    created_at: datetime


class DetectionJobResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    inspection_id: uuid.UUID
    model_version: str
    rag_version: str
    status: str
    error_message: str | None = None
    started_at: datetime | None = None
    completed_at: datetime | None = None
    created_at: datetime

File: app/lib/models/inspection.dart
class Inspection {
  const Inspection({
    required this.id,
    required this.annotationDomain,
    required this.inspectorId,
    required this.reportFlagged,
    required this.createdAt,
    this.status,
    this.imageKey,
    this.thumbnailKey,
  });

  final String id;
  final String annotationDomain;
  final String? status;
  final String inspectorId;
  final bool reportFlagged;
  final DateTime createdAt;
  final String? imageKey;
  final String? thumbnailKey;

  factory Inspection.fromJson(Map<String, dynamic> json) => Inspection(
        id: json['id'] as String,
        annotationDomain: json['annotation_domain'] as String,
        inspectorId: json['inspector_id'] as String,
        reportFlagged: json['report_flagged'] as bool,
        createdAt: DateTime.parse(json['created_at'] as String),
        status: json['status'] as String?,
        imageKey: json['image_key'] as String?,
        thumbnailKey: json['thumbnail_key'] as String?,
      );
}
